tests/unittest.py

```diff
@@ -102,7 +103,14 @@ class TestCase(unittest.TestCase):
     use_memory_storage: bool = USE_MEMORY_STORAGE
     seed_config: Optional[int] = None
 
+    def _runFixturesAndTest(self, result):
+        try:
+            return super()._runFixturesAndTest(result)
+        finally:
+            self.assertEqual(0, Simulator._patches_rc)
+
```

Comment:
Is this necessary? Isn't it the same as asserting in the `tearDown()`?

Reply:
The difference is that any class can override the `tearDown()` and prevent calling `super().tearDown()`. Besides that, it's exactly the same. Maybe I should keep only one of them.

Reply:
Another difference is that, even if `super.tearDown()` method is called, it must be called after all simulators have been stopped. Otherwise the assert will fail.

Reply:
Makes sense. I'm fine keeping both, then. I'm resolving this